db/large_data_handler.cc

```diff
@@ -90,6 +102,13 @@ future<> large_data_handler::maybe_delete_large_data_entries(sstables::shared_ss
             return delete_large_data_entries(*schema, std::move(filename), db::system_keyspace::LARGE_PARTITIONS);
         });
     }
+    future<> partitions_with_too_may_rows = make_ready_future<>();
+    entry = sst->get_large_data_stat(sstables::large_data_type::rows_in_partition);
+    if (entry && entry->above_threshold) {
+        partitions_with_too_may_rows = with_sem([schema, filename, this] () mutable {
+            return delete_large_data_entries(*schema, std::move(filename), db::system_keyspace::LARGE_PARTITIONS);
```

Comment:
I think this would cause an extraneous warning if we try to delete both due to having too many rows AND for having a large partition.

It'd a better to set `large_partitions` instead.

I see that you forgot to add `partitions_with_too_may_rows` to the `when_all` call below at the bottom of the function, so it wan't applied, so the fix to set `large_partitions` here makes even more sense...